### PR TITLE
  Automation/7586-slack-alert-when-approve-command-fails

### PR DESCRIPTION
**Pull request by IsraelleHub**
## Description of Change
As VA Mobile App, I want a proactive Slack alert to indicate when the approve command workflow failed so that I am aware of the issue proactively vs having to manually search or have Slack queries.

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by narin** [comment]
@IsraelleHub Is there a reason why we're creating new workflow here instead of adding this as an additional step on `release_pull_request.yml` that runs `if: failure()`? (See [documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-status-check-functions))

Either way, we should be able to test this once it's been merged by having one of the admins comment `/approve SOME_INVALID_VERSION` on a ticket. As you can see on [line 46 of `release_pull_request.yml`](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/4b7a0dcda5ed9865fc8f6b1357ac1b78e3882384/.github/workflows/release_pull_request.yml#L46), the workflow will try to run `git checkout` with the provided version. It should fail with with an invalid version since the branch does not exist.

File: .github/workflows/release_pull_request.yml [hunk 1]
```diff
@@ -67,4 +67,16 @@ jobs:
...
+        env:
+          SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+          channel_name: va-mobile-build-alerts
+          message: 'The `/approve` command for release: ${{inputs.version}} has failed. See: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|workflow run> for results.'
+        run: |
+            curl -X POST \
+                 -H "Authorization: Bearer $SLACK_API_TOKEN" \
+                 -H "Content-Type: application/json" \
+                 -d "{\"channel\":\"$channel_name\",\"text\":\"$message\"}" \
+                 https://slack.com/api/chat.postMessage
```
Comment (theodur) [inline .github/workflows/release_pull_request.yml:82]: We have the [start_slack_thread](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/.github/workflows/start_slack_thread.yml) reusable workflow that can be used for this

Comment (IsraelleHub) [reply .github/workflows/release_pull_request.yml]: done. Please check it out!

File: .github/workflows/release_pull_request.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ on:
         description: 'Version Number (eg. v1.1.0)'
         type: string
         required: true
+
```
Comment (narin) [inline .github/workflows/release_pull_request.yml:10]: Unneeded whitespace

File: .github/workflows/release_pull_request.yml [hunk 3]
```diff
@@ -15,14 +16,13 @@ on:
       devPrUrl:
         value: ${{jobs.create_prs.outputs.devPrUrl}}
       releaseHash:
-        value: ${{jobs.create_prs.outputs.releaseHash}}
+        value: ${{jobs.create_prs.outputs.releaseHash}}
```
Comment (narin) [inline .github/workflows/release_pull_request.yml:19]: Unneeded whitespace

File: .github/workflows/release_pull_request.yml [hunk 4]
```diff
@@ -15,14 +16,13 @@ on:
...
       releaseHash:
-        value: ${{jobs.create_prs.outputs.releaseHash}}
+        value: ${{jobs.create_prs.outputs.releaseHash}}     
   workflow_dispatch:
     inputs:
       version:
         description: 'Version Number (eg. v1.1.0)'
         required: true
         type: string
-
```
Comment (narin) [inline .github/workflows/release_pull_request.yml:25]: Would maybe add this line break back in to have separation between `on:` and `jobs:`

File: .github/workflows/release_pull_request.yml [hunk 5]
```diff
@@ -67,4 +67,14 @@ jobs:
...
+          
+  start_slack_thread:
+    name: Start Slack thread
+    if: ${{ failure() }}
+    needs: create_prs
+    uses: ./.github/workflows/start_slack_thread.yml
+    secrets: inherit
+    with:
+      channel_name: va-mobile-build-alerts
+      message: 'The `/approve` command for release: ${{inputs.version}} has failed. See: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|workflow run> for results.'
```
Comment (narin) [inline .github/workflows/release_pull_request.yml:79]: I think this message would read better without the colons `:`.

**Review by narin — CHANGES_REQUESTED** [review]
Some minor nitpicking regarding whitespace and slack message but overall looks good.

**Review by narin — APPROVED** [review]

**Review by timwright12 — APPROVED** [review]

**Pull request merged**